osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/RsaEncryptionProvider.java

```diff
@@ -0,0 +1,73 @@
+package org.opensmartgridplatform.shared.security.providers;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import javax.crypto.Cipher;
+
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+
+public class RsaEncryptionProvider extends AbstractEncryptionProvider implements EncryptionProvider {
+
+    public static final String ALG = "RSA";
+    public static final String ALGORITHM = "RSA/ECB/PKCS1Padding";
+
+    private Key publicKey;
+    private Key privateKey;
+
+    public RsaEncryptionProvider(File privateKeyStoreFile, File publicKeyStoreFile) {
+        try {
```

Comment:
I think I wouldn't do this in the constructor. You could create beans and inject the Key objects in here